File: WeatherMonitoring/WeatherReport.cs
namespace WeatherMonitoring;

public class WeatherReport
{
	public required string Location { get; set; }
	public long Time { get; set; }
	public float Temp { get; set; }
	public int Humidity { get; set; }
	public int Pressure { get; set; }
	public float WindSpeed { get; set; }
	public float Latitude { get; set; }
	public float Longitude { get; set; }
	public required Weather Weather { get; set; }
}

public class Weather
{
	public int WeatherId { get; set; }
	public required string Main { get; set; }
	public string? Description { get; set; }
}

File: FileProcessing/models/WeatherReport.go
package models

type WeatherReport struct {
	Location  string  `json:"location"`
	Time      int64   `json:"time"`
	Temp      float32 `json:"temp"`
	Humidity  int     `json:"humidity"`
	Pressure  int     `json:"pressure"`
	WindSpeed float32 `json:"windSpeed"`
	Latitude  float32 `json:"latitude"`
	Longitude float32 `json:"longitude"`
	Weather   Weather `json:"weather"`
}

type Weather struct {
	WeatherId   int     `json:"weatherId"`
	Main        string  `json:"main"`
	Description *string `json:"description"`
}
